### PR TITLE
Client core interface implementation

### PR DESCRIPTION
**Pull request by Daniele-Tentoni**
First client core interface implementation. Missing prolog functions.

### CODE REVIEW

File: core/src/main/scala/it/parttimeteam/GameState.scala [hunk 1]
```diff
@@ -2,14 +2,19 @@ package it.parttimeteam
...
  *
- * @param players list of player.
  * @param deck    deck game.
  * @param board   board.
+ * @param players list of player.
  */
-case class GameState(players: List[Player], deck: Deck, board: Board) {
+case class GameState(deck: Deck, board: Board, players: Seq[Player]) {
   // TODO: Implement utilities functions.
+  def getP(id: PlayerId): Option[Player] = players.find(p => p.id equals id)
```
Comment (MatteoCaval) [inline core/src/main/scala/it/parttimeteam/GameState.scala:16]: what is p?

File: core/src/main/scala/it/parttimeteam/core/collections/Deck.scala [hunk 2]
```diff
@@ -54,7 +54,7 @@ object Deck {
    *
    * @return List of cards.
    */
-  def empty: Deck = Deck(List[Card]())
+  def empty: Deck = Deck(Nil)
```
Comment (MatteoCaval) [inline core/src/main/scala/it/parttimeteam/core/collections/Deck.scala:57]: why not List.empty?

File: core/src/main/scala/it/parttimeteam/GameState.scala (hunk 1 above)
Comment (Daniele-Tentoni) [reply core/src/main/scala/it/parttimeteam/GameState.scala]: Is a player.

**Review by MatteoCaval — DISMISSED** [review]

File: core/src/main/scala/it/parttimeteam/core/collections/Deck.scala (hunk 2 above)
Comment (Daniele-Tentoni) [reply core/src/main/scala/it/parttimeteam/core/collections/Deck.scala]: Because Nil is shorter. Now I fix this.

**Review by MatteoCaval — APPROVED** [review]

**Pull request merged**